chain/x/zoracle/internal/types/request.go

```diff
@@ -1,32 +1,39 @@
 package types
 
 import (
-	"fmt"
-	"strings"
+	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
-// Request is a type to store detail of request
+// Request is a type to store detail of request.
 type Request struct {
-	CodeHash    []byte `json:"codeHash"`
-	Params      []byte `json:"params"`
-	ReportEndAt uint64 `json:"reportEnd"`
+	OracleScriptID           int64            `json:"oracleScriptID"`
+	Calldata                 []byte           `json:"calldata"`
+	ValidatorList            []sdk.ValAddress `json:"validatorList"`
+	SufficientValidatorCount int64            `json:"sufficientValidatorCount"`
+	SubmittedValidatorList   []sdk.ValAddress `json:"submittedValidatorList"`
+	RequestedBlock           int64            `json:"requestedBlock"`
+	RequestedTime            int64            `json:"requestedTime"`
+	ExpiredBlock             int64            `json:"expiredBlock"`
+	IsResolved               bool             `json:"isResolved"`
```

Comment:
As we already talked, I think we'll need to have a way to track the list of external data requests. Each will have the external ID, the data source ID, and calldata.